pages/index.tsx

```diff
@@ -1,36 +1,17 @@
-import { Box, Button, Container, Typography } from '@mui/material';
+import { Box, Button, Container, List, ListItem, Typography } from '@mui/material';
 import jwt from 'jsonwebtoken';
 import { NextPage } from 'next';
-import React, { useState, useEffect } from 'react';
+import React from 'react';
 import Cookies from 'universal-cookie';
 
 import createAxiosInstance from 'client/lib/axios';
 import { getDiscordOauthUrl } from 'client/lib/oauth';
+import { User } from 'server/models';
 import { AUTH_COOKIE_NAME } from 'shared/constants';
 
-interface User {
-  id: number;
-  discord_name: string;
-}
-
-interface Props {
-  isAuthed: boolean;
-}
-
-const Index: NextPage<Props> = props => {
+const Index: NextPage<{ isAuthed: boolean; users: User[] }> = props => {
```

Comment:
@timmyichen Can I get some clarification on how you wanted the User to be referenced in this?

Reply:
Do we have any types representing the a user, but only the fields the client will see?  That's what we should be using here (we should avoid doing any kind of server import on the client)

Reply:
`UserProfile`, though it's in `server/types/User` [[Link](https://github.com/timmyichen/dev-directory/blob/75cc63466ce7b3cf4fcbb607c3b69d1b92054c74/server/types/User.ts#L3)]

Reply:
guh.  more reason to use sequelize-typescript imo.  i'd be fine with just typing this as `any` and adding a TODO to type it once sequelize-typescript is added

Reply:
> Do we have any types representing the a user, but only the fields the client will see? That's what we should be using here (we should avoid doing any kind of server import on the client)

What's wrong with importing types from the server?

Reply:
Because the client should be assumed to _not_ have direct access to server code and resources.

The risk is that server code is now bundled with the client, exposing how the app works and allowing clients to get a sense of how the backend works and find vulnerabilities. Obviously this doesn't matter since it's open source, but it's general good practice to follow.

I suspect it's not actually breaking anything right now because `User` is only used as a type, which gets stripped out on compilation, but if we were to use `User` in any other non-type capacity it will probably break the app.

Reply:
How will sequelize-typescript give the client the type without having the same issue?

Reply:
sequelize-typescript allows us to define an interface (which can be used by both client and server, and can live in the shared directory) and create a class that is coupled with the interface, allowing us to have a single source of truth for what fields exist for a user.